pkg/apis/pipeline/v1/param_types.go

```diff
@@ -131,27 +131,28 @@ const (
 // AllParamTypes can be used for ParamType validation.
 var AllParamTypes = []ParamType{ParamTypeString, ParamTypeArray, ParamTypeObject}
 
-// ArrayOrString is modeled after IntOrString in kubernetes/apimachinery:
+// ParamValues is modeled after IntOrString in kubernetes/apimachinery:
 
-// ArrayOrString is a type that can hold a single string or string array.
+// ParamValues is a type that can hold a single string, string array, or string map.
 // Used in JSON unmarshalling so that a single JSON field can accept
 // either an individual string or an array of strings.
-// TODO (@chuangw6): This struct will be renamed or be embedded in a new struct to take into
-// consideration the object case after the community reaches an agreement on it.
-type ArrayOrString struct {
-	Type      ParamType `json:"type"` // Represents the stored type of ArrayOrString.
+type ParamValues struct {
```

Comment:
Thanks! Another idea of this PR is to collect the ideas of new Namings, we have discussed this in this Monday's api working group

Reply:
I think `ParamValue` is a good choice. `ParamValues` is definitely not so much, since the struct only technically holds the value for a single param, so the plural is misleading.

Reply:
Sure, I will change to ParamValue